modules/loader/src/main/scala/com/snowplowanalytics/snowplow/postgres/config/LoaderConfig.scala

```diff
@@ -127,21 +131,35 @@ object LoaderConfig {
           }.or(deriveConfiguredDecoder[Retrieval])
         }
       }
+
+      case class CheckpointSettings(maxBatchSize: Int, maxBatchWait: FiniteDuration) {
+        def unwrap: KinesisCheckpointSettings =
+          KinesisCheckpointSettings(maxBatchSize, maxBatchWait).toOption.get
```

Comment:
`Option.get` smells bad and it also seems possible to hit `None.get` here.

`KinesisCheckpointSettings.apply` will return a `Left` when either `maxBatchSize <= 0` or `maxBatchWait <= 0 milliseconds` and current decoder will accept negative maxBatchSize (I didn't test it by running but it seems so).

Decoder could be made stricter to satisfy `KinesisCheckpointSettings.apply`'s requirements so that our app's inner world would be safer to work with. Even in that case I'd avoid `Option.get` if possible.

Reply:
I just saw the custom decoder below 👍